### ISSUE TITLE
I have some problems when training model by myself

### ISSUE THREAD

**Issue by dong1015323606** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#4]
Thank you for sharing the code first. But when I read your code (Train.py), I found some things that confused me, as follows:

1. I don't understand the function of "-20" and "20" in these two lines of code
    ` op = tf.image.resize_images(tf.nn.relu(op * -20), [self._left_input_batch.get_shape()[1].value, self._left_input_batch.get_shape()[2].value]) ` lines in 70 of Train.py

    `u5 = tf.image.resize_images(V6, [image_height // scales[5], image_width // scales[5]]) * 20. / scales[5]` lines in 282 of Train.py

    `rescaled_prediction = tf.image.resize_images(self._get_layer_as_input('final_disp'), [image_height, image_width]) * -20.` lines in 371 of Train.py

2. When I trained on the Monkaa data set, I found that the value of loss always jumped back and forth, and the model did not converge.  After many times of failures, I found that some code confused me in the process of network construction. 

`rescaled_prediction = tf.image.resize_images(self._get_layer_as_input('final_disp'), [image_height, image_width]) * -20.` lines in 371 of Train.py

   In my understand, this code pad the final disparity map to the size of groundtruth. Then use the padded map compute the loss. I think it is easier to converge by reducing the size of groudtruth in the process of computing the loss.  I would appreciate it if you could answer my questions. Thank you very much.

**Comment by AlessioTonioni** [comment]
The values predicted by the network should be multiplied by -20 to get the real disparity since it was trained by scaling the gt disparity by -20. 
We predict negative disparity following the guidelines of Dispnet (https://github.com/lmb-freiburg/dispnet-flownet-docker) and divide the GT disparity by 20 following the guidelines of PWCNet for optical flow (https://github.com/NVlabs/PWC-Net/tree/master/PyTorch)

> I think it is easier to converge by reducing the size of groudtruth in the process of computing the loss

In our tests, this strategy is worse than scaling the predictions to full res and computing the loss at that resolution. In the end, at test time, what we want is good full-scale disparity, so that's what we try to optimize during training.

**Comment by dong1015323606** [comment]
Thank you  for your answer, it's very helpful for me. But I would like to ask if you have trained MADNet on the Monkaa dataset. 
   I think MADNet has an excellent structure, but when training, it always fails to converge, which makes me confused. I use the loss function as sum_l1, the learning rate is 0.0001, and the image size is 540*960. 
   By the way, I tried two loss weights included  "0.1 0 0 0 0 0" and "0.001 0.005 0.01 0.02 0.08 0.32". And I used about 9000 pairs of images training MADNet from the Monkaa Dataset. Both the size of the batch and the number of epochs are default (4 and 50). 
   Do you see anything wrong with what I said above. I would appreciate it if you could answer my questions. Thank you very much.

**Comment by dong1015323606** [comment]
I'm sorry that I made a mistake. First, the loss weights mentioned in paper should be 0,  0.005,  0.01,  0.02, 
 0.08,  0.32. Second, the loss function used in training should be sum_L2. I will try to train again after change these parameter.

**Comment by AlessioTonioni** [comment]
> But I would like to ask if you have trained MADNet on the Monkaa dataset.
No never tried. The base model was trained on Flying Things 3D

> Finally, I would like ask you the size of crop patch when training initial model on the FlyingThings3D
384 x 768

> the loss function used in training should be sum_L2
No actually we used sum_l1

**Comment by dong1015323606** [comment]
Thank you for your answer, I will verify what happens to the experimental results when these parameters change.
